packages/run-protocol/src/psm/psm.js

```diff
@@ -0,0 +1,170 @@
+  const { zcfSeat: feePool } = zcf.makeEmptySeatKit();
+  const { zcfSeat: stage } = zcf.makeEmptySeatKit();
+
+  const assertUnderLimit = anchorAmount => {
+    assert(
+      AmountMath.isGTE(gov.getMintLimit(), anchorAmount),
+      X`Request would exceed mint limit`,
+    );
+  };
+
+  const giveStable = (seat, given, wanted = emptyAnchor) => {
+    const fee = ceilMultiplyBy(given, gov.getGiveStableRate());
+    const afterFee = AmountMath.subtract(given, fee);
+    const maxAnchor = AmountMath.make(anchorBrand, afterFee.value);
+    // TODO this prevents the reallocate from failing. Can this be tested otherwise?
+    assert(
+      AmountMath.isGTE(maxAnchor, wanted),
+      X`wanted ${wanted} is more then ${given} minus fees ${fee}`,
+    );
+    stageTransfer(seat, stage, { In: afterFee }, { Stable: afterFee });
+    stageTransfer(seat, feePool, { In: fee }, { Stable: fee });
+    stageTransfer(anchorPool, seat, { Anchor: maxAnchor }, { Out: maxAnchor });
+    zcf.reallocate(seat, anchorPool, stage, feePool);
+    stableMint.burnLosses({ Stable: afterFee }, stage);
+  };
+
+  const wantStable = (seat, given, wanted = emptyStable) => {
+    const anchorAfterTrade = AmountMath.add(
+      anchorPool.getAmountAllocated('Anchor', anchorBrand),
+      given,
+    );
+    assertUnderLimit(anchorAfterTrade);
+    const asStable = AmountMath.make(stableBrand, given.value);
+    const fee = ceilMultiplyBy(asStable, gov.getWantStableRate());
+    const afterFee = AmountMath.subtract(asStable, fee);
+    assert(
+      AmountMath.isGTE(afterFee, wanted),
+      X`wanted ${wanted} is more then ${given} minus fees ${fee}`,
+    );
+    stableMint.mintGains({ Stable: asStable }, stage);
+
+    stageTransfer(seat, anchorPool, { In: given }, { Anchor: given });
+    stageTransfer(stage, seat, { Stable: afterFee }, { Out: afterFee });
+    stageTransfer(stage, feePool, { Stable: fee });
+    try {
+      zcf.reallocate(seat, anchorPool, stage, feePool);
+    } catch (e) {
+      // TODO reallocate should guarantee that this case cannot happen
+      stableMint.burnLosses({ Stable: asStable }, stage);
+      throw e;
+    }
+  };
+
+  const swapHook = seat => {
+    assertProposalShape(seat, {
+      give: { In: null },
+    });
+    const {
+      give: { In: given },
+      want: { Out: wanted } = { Out: undefined },
+    } = seat.getProposal();
+    if (given.brand === stableBrand) {
+      giveStable(seat, given, wanted);
+    } else if (given.brand === anchorBrand) {
+      wantStable(seat, given, wanted);
+    } else {
+      throw Error(`unexpected brand ${given.brand}`);
+    }
+    seat.exit();
+  };
+  const makeSwapInvitation = () => zcf.makeInvitation(swapHook, 'swap');
+
+  // Eventually the reward pool will live elsewhere. For now it's here for
+  // bookkeeping. It's needed in tests.
+  const getRewardAllocation = () => feePool.getCurrentAllocation();
+
+  const publicFacet = Far('Parity Stability Module', {
+    getCurrentLiquidity: () =>
```

Comment:
I don't think I'd call this 'Liquidity'. I think 'collateral', 'balance', or 'reserves' would make more sense to me.

Reply:
I like 'balance' and 'reserve'. It might make sense to change all the "anchor" names to "reserve" as well, since that's the reserve asset for this contract.

Reply:
Were you going to change the names, or were you waiting for more discussioni?

Reply:
The git UI had several items visually collapsed so I missed them.  Fixed.

Reply:
Now it's `getPoolBalance`